kanidm_unix_int/tests/cache_layer_test.rs

```diff
@@ -514,31 +514,58 @@ fn test_cache_account_pam_allowed() {
         let fut = async move {
             cachelayer.attempt_online().await;
 
-            // Should fail
-            let a1 = cachelayer
-                .pam_account_allowed("testaccount1")
-                .await
-                .expect("failed to authenticate");
-            assert!(a1 == Some(false));
+            // ! TRACING BEGIN
+            let span = span!(Level::INFO, "test_cache_account_pam_allowed");
+            let _enter = span.enter();
+            // ? Seems like these subscribers will allow us to have json loggers
+            // ? and human-readable loggers at the same time, and select between
+            // ? them easily?
+
+            let subscriber = FmtSubscriber::builder()
+                .with_max_level(Level::INFO)
+                // .event_format(format::json()) // ! uncomment for JSON
+                .finish();
+            // ? This is specific to only applications-
+            // ? Ask if kanidm will ever be used as a library?
+            // ? I'm guessing not but double check
+
+            // Set a global subscriber (bad)
```

Comment:
Why is this bad?

Reply:
It's bad because I was using it as a hack to make my code work, when I really should have been using scoped `subscriber`s. But I think we should end up using it as a safety net for default logs, probably with JSON formatting so we can do easy lookups. For most logging that we want to look at, I think we should use scoped `subscriber`s with human-readable formatting and pretty colors :)

One question, where should I initialize the global `subscriber`? I think we want to do it as early as possible, but I don't know where the entry point is.